src/types.ts

```diff
@@ -5,6 +5,13 @@ export interface App {
   url: string;
 }
 
+export interface WorkspaceSpec {
+  name: string;
+  nodeTables?: string[];
+  edgeTables?: string[];
+  graphs?: string[];
+}
+
```

Comment:
This seems like it should go in the state definition module. This "spec" is actually a spec for the workspace state data right?

Also, why are `nodeTables`, `edgeTables`, and `graphs` optional? They should be required, defaulting to empty arrays, no?

Reply:
> This seems like it should go in the state definition module. This "spec" is actually a spec for the workspace state data right?

Yeah, I suppose a more appropriate name would be `WorkspaceState` or something, as it really just represents the state of the current workspace.

> Also, why are `nodeTables`, `edgeTables`, and `graphs` optional? They should be required, defaulting to empty arrays, no?

I don't think there's a good reason, I can make them required.

Reply:
0455145